test/src/test/java/org/corfudb/runtime/clients/LogUnitHandlerTest.java

```diff
@@ -180,35 +180,7 @@ public void writeRange() throws Exception {
     }
 
     @Test
-    public void readingTrimmedAddress() throws Exception {
-        byte[] testString = "hello world".getBytes();
-        final long address0 = 0;
-        final long address1 = 1;
-        client.write(address0, Collections.<UUID>emptySet(), null, testString, Collections.emptyMap()).get();
-        client.write(address1, Collections.<UUID>emptySet(), null, testString, Collections.emptyMap()).get();
-        LogData r = client.read(address0).get().getAddresses().get(0L);
-        assertThat(r.getType())
-                .isEqualTo(DataType.DATA);
-        r = client.read(address1).get().getAddresses().get(1L);
-        assertThat(r.getType())
-                .isEqualTo(DataType.DATA);
-
-        client.prefixTrim(new Token(0L, address0));
-        client.compact();
```

Comment:
Tests using the client directly should not be affected by the async processing in the server.
I can't reproduce this, but it might be failing because the client doesn't wait on prefix trim and compact. Maybe you can do:
client.prefixTrim(new Token(0L, address0)).get();
client.compact().get();